base/iterators.jl

```diff
@@ -1014,7 +1015,7 @@ _prod_axes1(a, A) =
     throw(ArgumentError("Cannot compute indices for object of type $(typeof(a))"))
 
 ndims(p::ProductIterator) = length(axes(p))
-length(P::ProductIterator) = prod(size(P))
+length(P::ProductIterator) = reduce(Checked.checked_mul, size(P); init=1)
```

Comment:
checked_length? This seems otherwise possibly rather slow

Reply:
Aha! Do you mean that I should leave `length` as-is, and instead make a new method `Checked.checked_length(P::ProductIterator)`?

Reply:
Is the length of a product iterator a performance-critical calculation?  It seems better to default to safety here, since overflow is very likely and performance doesn't seem to matter to much for this function.